api/src/main/java/module-info.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2022 International Business Machines Corp. and others
+ * 
+ * See the NOTICE file distributed with this work for additional information
+ * regarding copyright ownership. Licensed under the Apache License, 
+ * Version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ * 
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+module jakarta.batch {
+
+    // Used in BatchRuntime
+    requires java.logging;
+
+    // Optional - note this transitively requires jakarta.inject
+    requires static jakarta.cdi;
+
```

Comment:
It's OK that jakarta.cdi is option. But shouldn't we also add an explicit dependency on jakarta.inject, since that one is really required all the time to support batch properties injection?
```suggestion
    requires jakarta.inject;

```

Reply:
I apologize since though I brought up this issue, I don't think my comments have kept up with the code I pushed here.

I see your point and we could make this change (I'd be OK with that).

One point I started to make: it is possible to have an application that just starts and monitors, etc. jobs via JobOperator that does NOT itself require 'inject'.   

So from a purely minimalist point of view I think the case could be made that it's "optional", strictly speaking.

However... jakarta.inject is a small API that doesn't drag in other dependencies while jakarta.cdi is a good bit bigger.

If you think a more reasonable compromise is to include a runtime dependency for jakarta.inject, and keep the jakarta.cdi dependency as static/optional... I'd believe you.   I don't have a lot of real-world experience with JPMS to be honest.

@rmannibucau had commented too on this point.  Reading that now, I think he more agreed with you.

Reply:
Let's see it one level higher: Scott is right inject is optional for end user but if the end user needs it, it will add it cause its code depends on it so let's ensure it is statically required only maybe? it would also align with the API itself which has both jars as provided (https://github.com/eclipse-ee4j/batch-api/blob/master/api/pom.xml#L33)

Reply:
My thinking was along what the spec requires from the implementation. And DI support is mandatory, while CDI support is optional, am I wrong? Therefore, since DI needs to be in the impl, it should be mandatory. And also transitive so that it's available to the user even if they only require batch and not jakarta.inject. Now I realize that required modules aren't automatically transitive, so I'd suggest that we add the following:

```suggestion
    requires transitive jakarta.inject;
    
```

I only suggest it because it seems as the purest solution for me (works like maven - you only need to require batch and you can also use @Inject without any other required module). I don't insist on it, we can leave it to the user to add the dependency on DI or revisit this in a later version of the spec.

Reply:
What if we do? 
> requires static transitive jakarta.inject;

This way we get a single dependency for the typical application with a batch job implementation (artifacts, etc.), and a user still has the option to build a batch operator/client without using **inject** or **cdi**.

Reply:
That sounds good, I agree with that.